### PR TITLE
Add Spring Data JPA to database

### PR DESCRIPTION
**Pull request by wilcorg**

### CODE REVIEW

File: src/main/java/org/gogame/server/domain/entities/UserEntity.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    @Column(length=64, nullable = false, unique = true)
+    private String nickname;
+
+    @Column(length=64, nullable = false)
+    private String password_hash;
+
+    @Column(length=128, nullable = false, unique = true)
+    private String email;
+
+    private java.sql.Timestamp join_date;
```
Comment (imnotpoz) [inline src/main/java/org/gogame/server/domain/entities/UserEntity.java]: don't use full paths if possible, import this

File: src/main/java/org/gogame/server/domain/entities/GameJournalEntity.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    private GameActionType actionType;
+
+    @Range(min = 0, max = 18)
+    private Integer turn_x;
+
+    @Range(min = 0, max = 18)
+    private Integer turn_y;
+
+    @Column(nullable = false)
+    private java.sql.Timestamp turn_date;
```
Comment (imnotpoz) [inline src/main/java/org/gogame/server/domain/entities/GameJournalEntity.java]: same here, import this

File: src/main/java/org/gogame/server/domain/entities/GameEntity.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Entity
+@Table(name = "game")
+public class GameEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "game_id_seq")
+    @SequenceGenerator(name = "game_id_seq", allocationSize = 1)
```
Comment (imnotpoz) [inline src/main/java/org/gogame/server/domain/entities/GameEntity.java:16]: do we need a custom generator? doesn't it increment by 1 by default?

File: src/test/java/org/gogame/server/repositories/GameJournalRepositoryIntegrationTests.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    @Autowired
+    private final GameRepository gameRepo;
+
+    @Autowired
+    private final UserRepository userRepo;
+
+    @Autowired
+    private final GameJournalRepository gameJournalRepo;
+    @Autowired
+    public GameJournalRepositoryIntegrationTests(GameRepository gameRepo,
```
Comment (imnotpoz) [inline src/test/java/org/gogame/server/repositories/GameJournalRepositoryIntegrationTests.java:29]: I'm pretty sure you don't need `@Autowired` on fields if it's on the constructor

File: src/main/java/org/gogame/server/domain/entities/GameEntity.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    @JoinColumn(name = "winner_id", referencedColumnName = "user_id")
+    private UserEntity winner;
+
+    @ManyToOne
+    @JoinColumn(name = "user_black_id", referencedColumnName = "user_id")
+    private UserEntity user_black;
+
+    @ManyToOne
+    @JoinColumn(name = "user_white_id", referencedColumnName = "user_id")
+    private UserEntity user_white;
```
Comment (imnotpoz) [inline src/main/java/org/gogame/server/domain/entities/GameEntity.java]: any reason for these being snake_case instead of camelCase?

**Review by imnotpoz — CHANGES_REQUESTED** [review]

File: src/main/java/org/gogame/server/domain/entities/GameEntity.java (hunk 3 above)
Comment (wilcorg) [reply src/main/java/org/gogame/server/domain/entities/GameEntity.java:16]: Not an issue, default Spring Data JPA by default creates generator that increases id by 50
image

File: src/main/java/org/gogame/server/domain/entities/GameJournalEntity.java (hunk 2 above)
Comment (wilcorg) [reply src/main/java/org/gogame/server/domain/entities/GameJournalEntity.java]: fixed

File: src/main/java/org/gogame/server/domain/entities/UserEntity.java (hunk 1 above)
Comment (wilcorg) [reply src/main/java/org/gogame/server/domain/entities/UserEntity.java]: fixed

File: src/test/java/org/gogame/server/repositories/GameJournalRepositoryIntegrationTests.java (hunk 4 above)
Comment (wilcorg) [reply src/test/java/org/gogame/server/repositories/GameJournalRepositoryIntegrationTests.java:29]: Fixed everywhere, thanks

File: src/main/java/org/gogame/server/domain/entities/GameEntity.java (hunk 5 above)
Comment (wilcorg) [reply src/main/java/org/gogame/server/domain/entities/GameEntity.java]: Not an issue, Spring Data JPA uses property name as table column name by default, unless other name have not specified in `@Column()`

Comment (imnotpoz) [reply src/main/java/org/gogame/server/domain/entities/GameEntity.java]: isn't there any way to have the names camel case and make spring convert them into snake case when doing sql?

Comment (wilcorg) [reply src/main/java/org/gogame/server/domain/entities/GameEntity.java]: I've seen in offical docs that Spring uses camelCase for entity properties. You were right

**Pull request merged**